x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/aggregations/bucket/histogram/HistoBackedHistogramAggregator.java

```diff
@@ -86,9 +86,9 @@ public void collect(int doc, long owningBucketOrd) throws IOException {
                             } else {
                                 collectBucket(sub, doc, bucketOrd);
                             }
-                            // We have added the document already. We should increment doc_count by count - 1
+                            // We have added the document already. We should increment doc_count by count - docCountProvider
                             // so that we have added it count times.
-                            incrementBucketDocCount(bucketOrd, count - 1);
+                            incrementBucketDocCount(bucketOrd, count - docCountProvider.getDocCount(doc));
```

Comment:
Looks like we both fixed the same bug. I merged #74650, so will have a conflict here.